### DISCUSSION TITLE
Brightness scale changes when switching between two LG monitors

### DISCUSSION THREAD

**Discussion by devinbrown7** [MonitorControl/MonitorControl#960] [General]
At work I have the "Apple" LG monitor (24MD4KL-B) and at home I have a different LG monitor (LG 27UD69P-W). When I go between work and home MonitorControl stops working, and I have to restart the app to get some things working.
When I go from home to work the apple monitor goes black under 50%.
When I go from work to home, the on screen brightness controls stop working, and even a restart of the app won't fix this.

**Comment by the0neyouseek** [comment]
Hi @devinbrown7,
Your issues could be related a number of things, from cables to the monitor themselves.
- What type of cable do you use at home/work. ?
- Is your mac an Arm one (M1/Pro/Max) ?

For your monitor that goes black under 50%, try disabling the "Combine hardware and software dimming" option in the app settings and see if it helps.

**Comment by devinbrown7** [comment]
Thanks for the reply. Let me give some more thorough details.

Background info:
- MacBook Pro with the M1 Pro chip
- Work: LG 24MD4KL-B w/ Apple native brightness support, connected via USB-C cable directly to the monitor's USB-C input
- Home: LG 27UD69P-W, connected via USB-C to HDMI cable.
- Using MonitorControl v4.0.2


When I first installed MonitorControl at home and used it on that monitor everything worked as expected. I need to have "Combine hardware and software dimming" enabled in order to get the black to full brightness range working like the built-in display.

However, when connect to my monitor at work, the brightness scale is off (going black at 50%). Restarting MonitorControl fixes this.

Then when I connect to the monitor at home, the keyboard brightness keys won't work, and restarting MonitorControl doesn't fix this issue. It was working before I went to work and connected to the Apple-supported LG monitor with the native brightness control.

I would expect to be able to switch monitors and have them both behave normally without having to restart MonitorControl.


It seems like connecting to the LG monitor at work has changed the way MonitorControl treats my LG monitor at home.

I could turn off "Combine hardware and software dimming" when I get to work, but restarting MonitorControl works too. And I want it to be enabled when I'm at home.

**Reply by waydabber** [reply]
Hi, can you please check the `Identifier` under `Displays` at work and at home? With Apple Silicon this ID (CGDirectDisplayID) is not as unique as it was with Intel but a simple number that is related to the connection's location in the IO registry. The problem is that MonitorControl stores preferences attached to this ID which can cause unexpected things especially if the two displays with very different characteristics are being connected to the same port in an alternating way. We should add an option to use the serial number at all times instead of this ID for associating preferences to the display (this was added as an option to BetterDummy because of this already and also MC has this but only for virtual displays like SideCar, Airplay as these displays can change their ID frequently).

_EDIT: sorry, what I wrote was a mess, tried to rephrase it lol._

**Reply by waydabber** [reply]
I added an issue regarding this https://github.com/MonitorControl/MonitorControl/issues/961

**Reply by waydabber** [reply]
Also see https://github.com/MonitorControl/MonitorControl/issues/792 - the problem was similar.

**Reply by devinbrown7** [reply]
For the work monitor, the Displays tab says: 
LG UltraFine
Identifier: 3
Control method: Hardware (Apple)

For the home monitor:
LG Ultra HD
Identifier: 3
Control method: Hardware (DCC)

Also, when going from the work monitor yesterday to the home monitor today, I didn't get the issue where the brightness scale is off. I think the occurrence of this issue might be related to unplugging the monitor while the computer is awake, as opposed to unplugging and replugging into the other monitor while the computer is asleep.

**Reply by waydabber** [reply]
Yeah. MonitorControl cannot really tell those displays apart because of the same IDs at the moment and since the two displays use a different control method this might cause extra issues. It is not clear for me how the 2021 MBP works - do you have the same ID if you use different ports to connect or is it always 3?

**Reply by the0neyouseek** [reply]
I just switched port to see and I have the same Id `3`

**Reply by devinbrown7** [reply]
I switched ports as well and the ID is still 3.

**Reply by waydabber** [reply]
All right, thanks, thats an important bit of info!

But... I was wrong, what I mentioned eariler should not cause this issue as I already did deal with it in v4.0.0 (I just forgot about it sorry, but I now checked the code). Plugging/unplugging while wake or not wake should not make a difference as both when a display is connected or there is a wake event, the displays are fully reconfigured anyway.

Btw you say that keyboard control works at work but not at home. Are you sure that at work it does not simply work because it is an Apple display and you use native keyboard control and the keys are not routed through MonitorControl at all? it is quite hard to tell the difference but you can tell by looking at the MC slider while changing the brightness with the keyboard - if the slider's change ins instantenous, then it is routed through MC, if the slider moves a bit slower, then MC is only reacting to the brightness change and reads the display's parameters but does not actually changes the brightness, instead it is managed by the OS.
